app/services/concerns/waste_carriers_engine/can_merge_finance_details.rb

```diff
@@ -12,7 +12,12 @@ def merge_finance_details
         registration.finance_details.orders << order
       end
 
+      transient_payments_array = []
```

Comment:
Why was it necessary to break this into two steps?

Reply:
The code used to be 
```
transient_registration.finance_details.payments.each do |payment|
   registration.finance_details.payments << payment
end
```

This is because the each loop was actually not working properly when called on `transient_registration.finance_details.payments`. The loop actually does not complete and I believe it's due to `transient_registration.finance_details.payments` being edited by the line inside the loop (payment can only belong to one payments collection at a time). The breaking up was a way to ensure each payment actually gets iterated on. I will add a comment to that effect

Reply:
But the line inside the loop is acting on the registration, not on the transient_registration?

Reply:
Yes the line is acting on the registration, but if you move a payment from `transient_registration.finance_details.payments` into `registration.finance_details.payments ` it has edited the `transient_registration.finance_details.payments` array during the iteration

Reply:
Based on the spec that I was trying to fix, it was not copying it, but moving it instead